### PR TITLE
{math}[foss/2023a] magma v2.7.2 w/ CUDA 12.1.1

### PR DESCRIPTION
**Pull request by surak**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by surak** [comment]
@Micket

File: easybuild/easyconfigs/m/magma/magma-2.7.2-foss-2023a-CUDA-12.1.1.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'pic': True, 'openmp': True}
+
+source_urls = ['https://icl.cs.utk.edu/projectsfiles/%(name)s/downloads/']
+sources = [SOURCE_TAR_GZ]
+patches = ['%(name)s-2.7.2_allow-all-sms.patch']
+checksums = [
+    {'magma-2.7.2.tar.gz': '729bc1a70e518a7422fe7a3a54537a4741035a77be3349f66eac5c362576d560'},
+    {'magma-2.7.2_allow-all-sms.patch': '1fe721092d0193cad7c669bd8ffaa7c093a2547967232123ffb03460742f7b98'},
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/m/magma/magma-2.7.2-foss-2023a-CUDA-12.1.1.eb:19]:
```suggestion
    {'magma-2.7.2_allow-all-sms.patch': '2996485844d7e199ea57eaebf59903d3514a03536f82b655ce30de480e28b828'},
```

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
skl-rockylinux-88 - Linux Rocky Linux 8.8, x86_64, Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz (skylake), 1 x NVIDIA NVIDIA RTX A4000, 535.104.05, Python 3.6.8
See https://gist.github.com/SebastianAchilles/1632bb49434cf58fc389ae79cb003718 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 535.54.03, Python 3.9.16
See https://gist.github.com/SebastianAchilles/0cf1a1b8aaa8a985e500597e495c1e15 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
skl-rockylinux-88 - Linux Rocky Linux 8.8, x86_64, Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz (skylake), 1 x NVIDIA NVIDIA RTX A4000, 535.104.05, Python 3.6.8
See https://gist.github.com/SebastianAchilles/06d7d81e0fbc65f4e5a9d8133aacb83f for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
jrc0870.jureca - Linux Rocky Linux 8.8, AArch64, ARM UNKNOWN (neoverse_n1), 2 x NVIDIA NVIDIA A100-PCIE-40GB, 535.54.03, Python 3.6.8
See https://gist.github.com/SebastianAchilles/5f52f770f1ee9b9ae7752bb47a2d5369 for a full test report.

File: easybuild/easyconfigs/m/magma/magma-2.7.2-foss-2023a-CUDA-12.1.1.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    ('CUDA', '12.1.1', '', SYSTEM),
+    ('UCX-CUDA', '1.14.1', versionsuffix),
+]
+
+# default CUDA compute capabilities to use (override via --cuda-compute-capabilities)
+cuda_compute_capabilities = ['5.0', '6.0', '7.0', '7.5', '8.0', '8.6', '9.0']
+
+# make sure both static and shared libs are built
+configopts = [
+    '-DMKL_ILP64=ON \
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/m/magma/magma-2.7.2-foss-2023a-CUDA-12.1.1.eb:35]: With the `foss` toolchain MKL can be used, but it is not the default backend for FlexiBLAS. That is why I think we should remove `MKL_ILP64` here.

```suggestion
```

Comment (Micket) [reply easybuild/easyconfigs/m/magma/magma-2.7.2-foss-2023a-CUDA-12.1.1.eb]: I pushed this change, but fix the string ' that needs to be put on the next line instead so i couldn't use the suggestion here.

**Review by Micket — APPROVED** [review]
lgtm (i think this probably can be brought down to gfbf level, it really doesn't make any difference i think..)

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Going in, thanks @surak!

**Comment by SebastianAchilles** [comment]
Going in, thanks @surak!

**Pull request merged**